project-base/storefront/components/Layout/Header/Cart/CartInHeader.tsx

```diff
@@ -38,97 +45,145 @@ export const CartInHeader: FC = ({ className }) => {
     const { url } = useDomainConfig();
     const [cartUrl] = getInternationalizedStaticUrls(['/cart'], url);
     const { removeFromCart, isRemovingFromCart } = useRemoveFromCart(GtmProductListNameType.cart);
+    const [isClicked, setIsClicked] = useState(false);
     const [isHovered, setIsHovered] = useState(false);
+    const isHoveredDelayed = useDebounce(isHovered, 200);
 
-    return (
-        <div
-            className={twMergeCustom('group relative lg:flex', className)}
-            onMouseEnter={() => setIsHovered(true)}
-            onMouseLeave={() => setIsHovered(false)}
-        >
-            {isCartFetchingOrUnavailable && (
-                <Loader className="absolute inset-0 z-overlay flex h-full w-full items-center justify-center rounded bg-backgroundMore py-2 opacity-50" />
-            )}
+    const { width } = useGetWindowSize();
+    const isDesktop = width > desktopFirstSizes.tablet;
 
-            <ExtendedNextLink
-                href={cartUrl}
-                tid={TIDs.header_cart_link}
-                className={twJoin(
-                    'min-w-24 hidden items-center gap-x-3 rounded h-12 pr-2 pl-4 no-underline transition-all hover:no-underline group-hover:rounded-b-none group-hover:shadow-lg lg:flex border',
-                    cart?.items.length ? nonEmptyCartTwClassName : emptyCartTwClassName,
-                )}
-            >
-                <span className="relative flex text-lg">
-                    <CartIcon className="w-6 lg:w-5" />
-                    {!!cart?.items.length && <CartCount>{cart.items.length}</CartCount>}
-                </span>
-                <span className="hidden text-sm font-semibold lg:block">
-                    {cart?.items.length
-                        ? formatPrice(cart.totalItemsPrice.priceWithVat, {
-                              explicitZero: true,
-                          })
-                        : t('Empty')}
-                </span>
-            </ExtendedNextLink>
+    const shouldDisplayTransportBar = cart?.remainingAmountWithVatForFreeTransport !== null && cart?.items.length;
 
+    return (
+        <>
             <div
-                className={twJoin(
-                    'pointer-events-none absolute top-full right-0 z-cart hidden origin-top-right scale-75 p-5 transition-all group-hover:pointer-events-auto group-hover:scale-100 group-hover:opacity-100 lg:block lg:rounded lg:rounded-tr-none lg:bg-background lg:opacity-0 lg:shadow-md',
-                    !cart?.items.length
-                        ? 'lg:flex lg:w-96 lg:flex-nowrap lg:items-center lg:justify-between'
-                        : 'lg:w-[510px]',
-                )}
+                className={twMergeCustom('group relative lg:flex z-aboveOverlay lg:pb-[10px] lg:mb-[-10px]', className)}
+                onMouseEnter={() => isDesktop && setIsHovered(true)}
+                onMouseLeave={() => isDesktop && setIsHovered(false)}
```

Comment:
How about using `useDebounce` to add a slight delay when the user hovers over the button?

Reply:
I used this for delay dark overlay to make it little bit animated. I already added delay for displaying hover popup. But we need do hover immediately. Try it on branch.